lms/djangoapps/discussion/static/discussion/js/views/discussion_board_view.js

```diff
@@ -0,0 +1,343 @@
+                    this.$('.forum-nav-browse-filter-input')
+                        .attr('aria-activedescendant', activeDescendantId)
+                        .val(activeDescendantText);
+                }
+            },
+
+            keyboardBinding: function(event) {
+                var key = event.which,
+                    $inputText = $('.forum-nav-browse-filter-input'),
+                    $filteredMenuItems = $('.forum-nav-browse-menu-item:visible'),
+                    filteredMenuItemsLen = $filteredMenuItems.length,
+                    $curOption = $filteredMenuItems.eq(0).find('.forum-nav-browse-title').eq(0),
+                    $activeOption, $prev, $next;
+
+                switch (key) {
+                case Constants.keyCodes.enter:
+                    $activeOption = $filteredMenuItems.find('.forum-nav-browse-title.is-focused');
+                    if ($inputText.val() !== '') {
+                        $activeOption.trigger('click');
+                        this.filterInputReset();
+                    }
+                    break;
+
+                case Constants.keyCodes.esc:
+                    this.toggleBrowseMenu(event);
+                    this.$('.forum-nav-browse-filter-input').val('');
+                    this.filterInputReset();
+                    $('.all-topics').trigger('click');
+                    break;
+
+                case Constants.keyCodes.up:
+                    if (this.selectedTopicIndex > 0) {
+                        this.selectedTopicIndex -= 1;
+                        if (this.isBrowseMenuVisible()) {
+                            $prev = $('.forum-nav-browse-menu-item:visible')
+                            .eq(this.selectedTopicIndex).find('.forum-nav-browse-title')
+                            .eq(0);
+                            this.filterEnabled = false;
+                            $curOption.removeClass('is-focused');
+                            $prev.addClass('is-focused');
+                        }
+                        this.selectOption($prev);
+                    }
+                    break;
+
+                case Constants.keyCodes.down:
+                    if (this.selectedTopicIndex < filteredMenuItemsLen - 1) {
+                        this.selectedTopicIndex += 1;
+                        if (this.isBrowseMenuVisible()) {
+                            $next = $('.forum-nav-browse-menu-item:visible')
+                                .eq(this.selectedTopicIndex).find('.forum-nav-browse-title')
+                                .eq(0);
+                            this.filterEnabled = false;
+                            $curOption.removeClass('is-focused');
+                            $next.addClass('is-focused');
+                        }
+                        this.selectOption($next);
+                    }
+                    break;
+
+                default:
+                    break;
```

Comment:
Breaking in default and then returning true invariably right after is a little strange. If the `return true` is needed, consider doing it wherever the switch currently breaks and then you won't need a final return.